skyplane/obj_store/gcs_interface.py

```diff
@@ -58,6 +58,7 @@ def map_region_to_zone(region) -> str:
                     f"No access to the Google Cloud Storage bucket '{self.bucket_name}', assuming bucket is in the 'us-central1-a' zone"
                 )
                 return "us-central1-a"
```

Comment:
not related to this PR, but why does this assume the bucket?

Reply:
Ah this is kinda frustrating and has to do with situations where someone copies data from a bucket that only has ListObject and GetObject permissions. An example is the Cloud TPU buckets (gs://cloud-tpu-test-datasets/fake_imagenet). Therefore, we don't know where the data is stored. We use `us-central1-a` to make a reasonable guess, but I guess in this situation we could potentially have chosen any GCP region including the destination GCP region if it were an intracloud transfer.

Reply:
@parasjain Would it make more sense to retry amongst a few regions before erroring out? This way users outside of this region would have the same experience as those inside.

Reply:
I made an issue to track this, I think ideally users should be able to configure fallback regions to let them control this.

Reply:
https://github.com/skyplane-project/skyplane/issues/663